pkg/buses/reconciler.go

```diff
@@ -531,10 +531,10 @@ func (r *Reconciler) syncSubscription(namespace string, name string) error {
 }
 
 func (r *Reconciler) createOrUpdateBus(bus *channelsv1alpha1.Bus) error {
-	if r.bus.GetObjectKind().GroupVersionKind().Kind != bus.Kind ||
-		bus.Namespace != r.bus.GetObjectMeta().GetNamespace() ||
+	if bus.Namespace != r.bus.GetObjectMeta().GetNamespace() ||
```

Comment:
the Kind needs to be checked because `r.bus` is a GenericBus and may not be a Bus. See https://github.com/knative/eventing/pull/298